envs/monkey_zoo/blackbox/config_templates/drupal.py

```diff
@@ -1,9 +1,10 @@
 from copy import copy
 
-from envs.monkey_zoo.blackbox.island_configs.base_template import BaseTemplate
+from envs.monkey_zoo.blackbox.config_templates.base_template import BaseTemplate
+from envs.monkey_zoo.blackbox.config_templates.config_template import ConfigTemplate
 
 
-class Drupal(BaseTemplate):
+class Drupal(ConfigTemplate):
     config_values = copy(BaseTemplate.config_values)
```

Comment:
In general, BaseTemplate could just be a constant instead of inheriting from ConfigTemplate. It's not worth holding up the merge, though.

Reply:
What are the benefits of that? The only difference is that now if there's a feature which requires only the base fields, we can simply inherit from BaseTemplate

Reply:
Then why not just inherit BaseTemplate like it did before?

BaseTemplate is used to sort of namespace a particular set of config values, but it's not a namespace, it's class. Then everything copies from it, which is mimicking a multiple inheritance scheme (since everything also inherits ConfigTemplate) without it being really explicit about what's going on.

If you did add a feature that only required the base fields, should it inherit BaseTemplate? Or should it follow the established pattern of copying from BaseTemplate? Nothing about the way it's designed now helps you make that decision.

Ultimately, it seems like a lot of confusion and all you get is a ConfigTemplate type hint. But since ConfigTemplate doesn't enforce anything about what is stored in the `config_value` dict, this type hint doesn't actually enforce anything; it just serves to obscure what a ConfigTemplate actually is.

I think an all-around better approach would be to make BaseTemplate a const, just use `dict` for the type hint and pass the appropriate `config_value` dict directly to whatever uses it. Since python modules are already objects, none of the class definitions, nor the inheritance scheme seem to add anything over just and populating defining config_value in each module.

All that being said, these classes are so small that confusion is minimal and it's easy to figure out what is going on, even if you're unsure why. It's not worth holding up this PR.

Reply:
Because "Composition over inheritance".

For a current infra, I agree - everything could be refactored into a dict, but I'm not sure that it would help with readability and making the codebase more extendable.